why_postgres.mdx

```diff
@@ -0,0 +1,137 @@
+
+If file attachments or full-text search or vector/hybrid search are important for your application, Xata will likely offer you a better platform to build on top than a classical DBaaS. And it's only the beginning, we are planning to have similar tightly-integrated solutions for time-series, intelligent caching, queuing, and more.
+
+## The implementation
+
+One key concept in the Xata implementation is the notion of _logical database branches_. A database can have multiple branches, and always has at least one, called `main` by default. A database branch is represented in Postgres as a [schema](https://www.postgresql.org/docs/current/ddl-schemas.html), and can be allocated on one of many Postgres clusters around the world, as well as moved from one cluster to another.
+
+This results in a level of decoupling between the logical databases and branches as experienced by the application, and the underlying Postgres instances where they are running. Because the clusters are already provisioned, logical databases and branches can be started instantly and allocated and moved depending on the business needs. This unlocks some interesting use cases, like sharding per tenant where every tenant is a branch, using a logical database for each micro-service, or automatically creating developer branches for each pull request. We use this logical separation in our free tier, where many small databases and branches share the same relatively large underlying Postgres cluster.
+
+![Database branches scheduled on various Postgres clusters](images/branches-clusters.png)
+
+In order to make this possible, we had to solve several challenges:
+
+- route traffic to the correct cluster based on metadata about the database branch
+- ensure that a branch level connection can only access that single branch
+- ensure that attackers can't call dangerous SQL functions
+- limit noisy-neighbor issues in the shared clusters
+
+#### SQL Proxy
+
+The key to the solution for most of these issues is our SQL Proxy technology. While there are many Postgres proxies (e.g. pgbouncer, pgcat, Supavisor) serving various goals, the Xata SQL Proxy not only speaks the Postgres wire protocol, but also understands the SQL being executed at a fairly deep level. It can, for example, know which tables are accessed, which columns are requested, which functions are being called with which parameters, and so on. This allows us to block actions that would be dangerous for the Postgres cluster. While security is mostly ensured via Postgres roles, the Proxy can enforce rate and concurrency limits on shared clusters (we don't do this for dedicated clusters).
```

Comment:
```suggestion
The key to the solution for most of these issues is our SQL Proxy technology. While there are many Postgres proxies (e.g. pgbouncer, pgcat, Supavisor) serving various goals, the Xata SQL Proxy not only speaks to the Postgres wire protocol, but also understands the SQL being executed at a fairly deep level. It can, for example, know which tables are accessed, which columns are requested, which functions are being called with which parameters, and so on. This allows us to block actions that would be dangerous for the Postgres cluster. While security is mostly ensured via Postgres roles, the Proxy can enforce rate and concurrency limits on shared clusters (we don't do this for dedicated clusters).
```

Reply:
OK I wasnt sure if it was meant to be like that or intentional so all good :)